\documentclass[journal]{IEEEtran}
\usepackage[cmex10]{amsmath}
\usepackage{cite}
\usepackage{amsmath,amssymb,amsfonts}
\usepackage{algorithmic}
\usepackage{graphicx}
\usepackage{textcomp}
%\usepackage{natbib}
\usepackage{comment}
\usepackage{changepage}
\usepackage{array}

%\usepackage[symbol]{footmisc}
%\usepackage{mathptmx}
%\usepackage{caption}
\usepackage{subcaption}
\usepackage{hyperref}
\usepackage{verbatim}
\usepackage{float}
\usepackage{multirow}
\usepackage{xr}
\externaldocument{supp}

\usepackage{makecell}
\renewcommand\theadalign{bc}
\renewcommand\theadfont{\bfseries}
\renewcommand\theadgape{\Gape[4pt]}
\renewcommand\cellgape{\Gape[4pt]}

%\renewcommand{\thefootnote}{\fnsymbol{footnote}}
\newcommand{\commentA}[1]{{ #1}} %\color{red}
\newcommand{\commentM}[1]{{ #1}} %\color{blue} 
\newcommand{\citep}[1]{\cite{#1}}
\newcommand{\citealp}[1]{\cite{#1}}
%%%%%%%for sqeezing table column from toufiue

\newcolumntype{L}[1]{>{\raggedright\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}
\newcolumntype{C}[1]{>{\centering\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}
\newcolumntype{R}[1]{>{\raggedleft\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}

%opening

\begin{document}
\title{Multi-objective formulation of MSA for phylogeny estimation 
	\\ \textit{\normalsize{(Do application-aware measures guide towards better phylogenetic tree?)}} }
\author{Muhammad Ali Nayeem,
	Md. Shamsuzzoha Bayzid,
	Atif Hasan Rahman,
	Rifat Shahriyar, and M. Sohel Rahman, \textit{Senior Member, IEEE}
\thanks{All the authors are with the Department of Computer Science and Engineering, Bangladesh University of Engineering and Technology, Dhaka 1205, Bangladesh. Corresponding author: M. Sohel Rahman (e-mail: msrahman@cse.buet.ac.bd).}
}
%\IEEEmembership{Senior Member, IEEE}}
%\address[1]{Department of CSE, BUET, Dhaka 1205, Bangladesh}
%\tfootnote{The first author is a recipient of the ICT PhD Fellowship administered by ICT Division, Government of People's Republic of Bangladesh.}


% The paper headers
\markboth{IEEE TRANSACTIONS ON CYBERNETICS}%
{Nayeem \MakeLowercase{\textit{et al.}}: Multi-objective formulation of MSA for phylogeny estimation}

%\corresp{Corresponding author: M. Sohel Rahman (e-mail: msrahman@cse.buet.ac.bd).}
\maketitle

\begin{abstract}
Multiple sequence alignment (MSA) is a basic step in many analyses in computational biology, including predicting the structure and function of proteins, orthology prediction and estimating phylogenies. The objective of MSA is to infer the homology among the sequences of chosen species. Commonly, the MSAs are inferred by optimizing a single function or objective. The alignments estimated under one criterion may be different from the alignments generated by other criteria, inferring discordant homologies and thus leading to different evolutionary histories relating the sequences. In recent past, researchers have advocated for the multi-objective formulation of MSA, to address this issue, where multiple conflicting objective functions are being optimized simultaneously to generate a set of alignments. However, no theoretical or empirical justification with respect to a real-life application has been shown for a particular multi-objective formulation. In this study, we investigate the impact of multi-objective formulation in the context of phylogenetic tree estimation. Employing multi-objective metaheuristics, we demonstrate that trees estimated on the alignments generated by multi-objective formulation are substantially better than the trees estimated on the alignments generated by the state-of-the-art MSA tools, including PASTA, MUSCLE, CLUSTAL, MAFFT, etc. We also demonstrate that highly accurate alignments with respect to popular measures like sum-of-pair (SP) score and total-column (TC) score do not necessarily lead to highly accurate phylogenetic trees. Thus, in essence, we ask the question of whether an application-aware (in this case phylogeny-aware) metric can guide us in choosing appropriate multi-objective formulations that can result in better phylogeny estimation. And we answer the question affirmatively through carefully designed extensive empirical study. As a by-product, we also suggest a methodology for primary selection of a set of objective functions for a multi-objective formulation based on the association with the resulting phylogenetic tree. In the sequel, we end up proposing a set of nonparametric objective functions (i.e., \{SimG, SimNG\}) that should be used by MSA methods if the goal is to produce high-quality phylogenetic trees.
\end{abstract}

\begin{IEEEkeywords}
%Enter key words or phrases in alphabetical order, separated by commas. For a list of suggested keywords, send a blank e-mail to keywords@ieee.org or visit \underline{http://www.ieee.org/organizations/pubs/ani\_prod/keywrd98.txt}
 Bioinformatics, Computational Biology, Evolutionary algorithm, Metaheuristics, Multiple sequence alignment, Multi-objective optimization, Phylogenetic tree.  
\end{IEEEkeywords}

%\titlepgskip=-15pt



\input{section_1_Introduction} 
\input{section_3_Methods}
\input{section_4_Results}
\input{section_6_Discussion}


\bibliographystyle{IEEEtran} 
\bibliography{document}

\begin{IEEEbiographynophoto}{Muhammad Ali Nayeem} received his B.Sc. and M.Sc. degrees in Computer Science and Engineering from the Department of Computer Science and Engineering (CSE), Bangladesh University of Engineering and Technology (BUET), Dhaka, Bangladesh in 2013 and 2016, respectively. 
	
%He is currently an Assistant Professor with the Department of CSE, BUET. His research interest includes metaheuristics and their application in real-life optimization problems, intelligent transportation systems, bioinformatics.
\end{IEEEbiographynophoto}

\begin{IEEEbiographynophoto}{Md. Shamsuzzoha Bayzid} is an Assistant Professor in the Department of Computer Science and Engineering (CSE) at Bangladesh University of Engineering and Technology (BUET).  He received his Ph.D. in Computer Science from the University of Texas at Austin (UT Austin) in 2016, and was supervised by Prof. Tandy Warnow. He completed the B. Sc. Engg. and M. Sc. Engg. degrees in Computer Science from BUET in 2008 and 2010, respectively. %Dr. Bayzidâ€™s research interest includes, but is not limited to, computational biology, theory and algorithms, and machine learning.	
\end{IEEEbiographynophoto}

\begin{IEEEbiographynophoto}{Atif Hasan Rahman} is an Assistant Professor in the Department of Computer Science and Engineering (CSE) at the Bangladesh University of Engineering and Technology (BUET). He completed his Ph.D. from the University of California, Berkeley in 2015 under the supervision of Lior Pachter. His research area is bioinformatics and computational biology and is currently focusing on statistical methods for genome assembly and analysis.% He was a recipient of Fulbright Science and Technology Fellowship in 2009.
\end{IEEEbiographynophoto}

\begin{IEEEbiographynophoto}{Rifat Shahriyar} is working as an Assistant Professor at Department of Computer Science and Engineering (CSE) of Bangladesh University of Engineering and Technology (BUET). He has completed Ph.D. in April 2015 from Research School of Computer Science in Australian National University (ANU). His Ph.D. under the supervision of Steve Blackburn and Kathryn McKinley changed the way people think about reference counting and conservative garbage collection. %He has completed B.Sc. in May 2007 and M.Sc. in December 2009 from CSE, BUET. His research interests are memory management specially garbage collection, virtual machine, programming language implementation, and high performance computing.
\end{IEEEbiographynophoto}

\begin{IEEEbiographynophoto}{M. Sohel Rahman} is a Professor of the CSE department of BUET. He had previously worked as a Visiting Senior Research Fellow of King's College London. He is a Senior Member of both IEEE and ACM and a member of American Mathematical Society (AMS) and London Mathematical Society (LMS). He is also a Peer-review Associate College Member of EPSRC, UK. Dr. Rahman's research interest includes, but is not limited to, theory and algorithms, stringology, metaheuristics, applied informatics, computational biology and bioinformatics. 
	%Among his highly cited results are the work on high dimensional Knapsack problems, algorithms on different variants of sequence alignment problems, data structures for different variants of string and sequence matching and sufficient conditions for Hamiltoninicity and metaheuristics solutions for hard real-life problems in different branches of science and engineering. Dr. Rahman regularly writes reviews at Mathematical Review and ACM Computing Review.
\end{IEEEbiographynophoto}

%\EOD
\vfill

\end{document}
